base/compiler/abstractinterpretation.jl

```diff
@@ -981,8 +1003,16 @@ function abstract_call_known(interp::AbstractInterpreter, @nospecialize(f),
             ft = argtype_by_index(argtypes, 3)
             (itft === Bottom || ft === Bottom) && return CallMeta(Bottom, false)
             return abstract_apply(interp, itft, ft, argtype_tail(argtypes, 4), sv, max_methods)
+        elseif f === invoke
+            ft = widenconst(argtype_by_index(argtypes, 2))
```

Comment:
Should check argument count.

Reply:
argtype_by_index will check and also handle the `Vararg` case correctly (similar to the apply case above).